framework/SupervisedLearning/DynamicModeDecompositionControl.py

```diff
@@ -0,0 +1,406 @@
+        construct a time-dependent surrogate model based on Dynamic
+        Mode Decomposition with Control (ref. \cite{proctor2016dynamic}).
+        In addition to perform a ``dimensionality reduction regression'' like DMD, this surrogate will
+        calculate the state-space representation matrices A, B and  C in a discrete time domain:
+        \begin{itemize}
+          \item $x[k+1]=A*x[k]+B*u[k]$
+          \item $y[k+1]=C*x[k+1]$
+        \end{itemize}
+
+        In order to use this Reduced Order Model, the \xmlNode{ROM} attribute
+        \xmlAttr{subType} needs to be set equal to \xmlString{DMDC}.
+        \\
+        Once the ROM  is trained (\textbf{Step} \xmlNode{RomTrainer}), its
+        parameters/coefficients can be exported into an XML file
+        via an \xmlNode{OutStream} of type \xmlAttr{Print}. The following variable/parameters can be exported (i.e.
+        \xmlNode{what} node
+        in \xmlNode{OutStream} of type \xmlAttr{Print}):
+        \begin{itemize}
+          \item \xmlNode{rankSVD}, see XML input specifications above
+          \item \xmlNode{actuators}, see XML input specifications above
+          \item \xmlNode{states}, XML node containing the list of system state variables (x)
+          \item \xmlNode{initStates}, XML node containing the list of system state variables
+                (x\_init) that are used for initializing the model in ``evaluation'' mode
+          \item \xmlNode{UNorm}, XML node containing the norminal values of actuators, which are the initial values in the
+                training data
+          \item \xmlNode{XNorm}, XML node containing the norminal values of state variables,
+                which are the initial values in the training data
+          \item \xmlNode{XLast}, XML node containing the last value of state variables,
+                which are the final values in the training data
+          \item \xmlNode{outputs}, XML node containing the list of system output variables (y)
+          \item \xmlNode{YNorm}, XML node containing the norminal values of output variables,
+                which are the initial values in the training data
+          \item \xmlNode{dmdTimeScale}, XML node containing the array of time scale in the DMD space
+          \item \xmlNode{Atilde},  XML node containing the A matrix in discrete time domain
+                (real and imaginary and part, matrix shape, and format note)
+          \item \xmlNode{Btilde}, XML node containing the B matrix in discrete time domain
+                (real and imaginary and part, matrix shape, and format note)
+          \item \xmlNode{Ctilde}, XML node containing the C matrix in discrete time domain
+                (real and imaginary and part, matrix shape, and format note)
+        \end{itemize}"""
+    specs.addSub(InputData.parameterInputFactory("dmdType", contentType=InputTypes.makeEnumType("dmd", "dmdType", ["dmd"]),
```

Comment:
dmdType is already defined in the DMD class, in this case, you will have two entries presented in the user manual

Reply:
This comes from the automatic documentation. Inheritance